### DISCUSSION TITLE
Does trino support passing command line arguments into sql scripts?

### DISCUSSION THREAD

**Discussion by changqing** [trinodb/trino#13374] [Q&A]
When using trino in batch mode, it is handy if it has this functionality.

**Comment by nineinchnick** [comment]
Can you show an example of what a command like that would look like?

**Reply by changqing** [reply]
For example, set variable on command line like Hive
![image](https://user-images.githubusercontent.com/2008743/182772590-a236eab5-ac23-4cec-bedd-6bc96df25714.png)

**Comment by hashhar** [comment]
There is an env variable `TRINO_PREPROCESSOR` which allows you to define a Java class which will be used to pre-process the CLI input. See https://github.com/trinodb/trino/blob/master/client/trino-cli/src/main/java/io/trino/cli/QueryPreprocessor.java
